pkgs/development/tools/java/visualvm/default.nix

```diff
@@ -1,27 +1,27 @@
-{ stdenv, fetchzip, lib, makeWrapper, jdk, gtk2, gawk }:
+{ stdenv, fetchzip, lib, makeWrapper, makeDesktopItem, jdk, gtk2, gawk }:
 
 stdenv.mkDerivation rec {
-  name = "visualvm-1.3.9";
+  version = "1.4";
+  name = "visualvm-${version}";
 
   src = fetchzip {
-    url = "https://github.com/visualvm/visualvm.src/releases/download/1.3.9/visualvm_139.zip";
-    sha256 = "1gkdkxssh51jczhgv680i42jjrlia1vbpcqhxvf45xcq9xj95bm5";
+    url = "https://github.com/visualvm/visualvm.src/releases/download/${version}/visualvm_${builtins.replaceStrings ["."] [""]  version}.zip";
+    sha256 = "0qpkal3yc8c8di4dmdglq54a4m7hciqximn5n6hwx2zqnfwx909z";
+  };
+
+  desktopItem = makeDesktopItem {
+      name = "visualvm";
```

Comment:
Any ideas on how to add the icon? There is an icon located at etc/visualvm.icns. Is it ok to just set icon to that location?

Reply:
I'm just wondering if an icns file is ok or does it need to be a png?

Reply:
Per https://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html the supported formats are png, xpm, and svg.

Reply:
I can't find a png file in the source. Would it be bad to convert the image to png using image magick?

Reply:
No idea, sorry